hpcbench/driver.py

```diff
@@ -595,6 +596,14 @@ def children(self):
             category = execution.get('category')
             if category != self.category:
                 continue
+            # Update `environment` key
+            yaml_env = self.config.get('environment')
+            if yaml_env:
```

Comment:
I don't think this is correct. this will get a an env from the yaml. If it finds it it will overwrite the benchmark env completely rather than updating it key by key.